### PR TITLE
lkl: introduce PIPE network backend for virtio device

### PR DESCRIPTION
**Pull request by motomuman**
This PR allows lkl to use anonymous pipe and named pipe as network channel.
This also includes tests for pipe network channel and small test fix for the android environment.

---
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/383)

### CODE REVIEW

**Comment by thehajime** [comment]
I worked the patchset with @motomuman (put Reviewed-by tags).

The basic idea might be somehow crazy: to use Unix pipes as a virtio device (backend).  The benchmark says it does not outperform any of existing LKL virtio-net backends, but it is definitely useful especially for testing purpose on localhost since it does not require root privilege at all.

File: tools/lkl/tests/net.sh [hunk 1]
```diff
@@ -81,7 +83,8 @@ if ! [ -z $DST ]; then
     ${SUDO} ip link set dev ${IFNAME} promisc off
 
     echo "== macvtap (LKL net) tests =="
-    ${SUDO} ip link add link ${IFNAME} name lkl_vtap0 type macvtap mode passthru
+    ${SUDO} ip link add link ${IFNAME} name lkl_vtap0 \
```
Comment (tavip) [inline tools/lkl/tests/net.sh]: Should we ignore it only for Android?

**Comment by tavip** [comment]
@lkl-jenkins: test this please

File: tools/lkl/tests/net.sh (hunk 1 above)
Comment (thehajime) [reply tools/lkl/tests/net.sh]: there is one more check with /dev/tap* right after this line, so it should be okay I think.

**Pull request merged**